// @evercourse/ever-modal/build/component/ConfirmModal.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRedux = require('react-redux');

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

var _index3 = require('../index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var confirmModal = function confirmModal(_ref) {
    var message = _ref.message,
        onSuccess = _ref.onSuccess,
        closeModal = _ref.closeModal,
        modalType = _ref.modalType;
    return _react2.default.createElement(
        _index2.default,
        { modalType: modalType, contentLabel: 'Confirm', className: 'modal-sm' },
        _react2.default.createElement(
            'div',
            { className: 'modal-body' },
            message
        ),
        _react2.default.createElement(
            'div',
            { className: 'modal-footer' },
            _react2.default.createElement(
                'button',
                { className: 'btn btn-danger confirm-btn', onClick: function onClick() {
                        onSuccess();
                        closeModal(modalType);
                    } },
                'Confirm'
            ),
            _react2.default.createElement(
                'button',
                { className: 'btn btn-secondary cancel-btn', onClick: function onClick() {
                        closeModal(modalType);
                    } },
                'Cancel'
            )
        )
    );
};

var mapStateToProps = function mapStateToProps(state, ownProps) {
    return {
        message: _index3.modalSelectors.getModalProps(state, ownProps.modalType).message,
        onSuccess: _index3.modalSelectors.getModalProps(state, ownProps.modalType).onSuccess
    };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
    return {
        closeModal: function closeModal(modalType) {
            dispatch(_index3.modalActions.hideModal(modalType));
        }
    };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(confirmModal);
