// @evercourse/ever-modal/build/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.modalReducer = exports.ModalRoot = exports.ConfirmModal = exports.OpenModalButton = exports.Modal = exports.modalSelectors = exports.modalActions = undefined;

var _reducer = require("./reducer");

var _reducer2 = _interopRequireDefault(_reducer);

var _actions = require("./actions");

var modalActions = _interopRequireWildcard(_actions);

var _selectors = require("./selectors");

var modalSelectors = _interopRequireWildcard(_selectors);

var _component = require("./component");

var _component2 = _interopRequireDefault(_component);

var _OpenModalButton = require("./component/OpenModalButton");

var _OpenModalButton2 = _interopRequireDefault(_OpenModalButton);

var _ConfirmModal = require("./component/ConfirmModal");

var _ConfirmModal2 = _interopRequireDefault(_ConfirmModal);

var _ModalRoot = require("./component/ModalRoot");

var _ModalRoot2 = _interopRequireDefault(_ModalRoot);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.modalActions = modalActions;
exports.modalSelectors = modalSelectors;
exports.Modal = _component2.default;
exports.OpenModalButton = _OpenModalButton2.default;
exports.ConfirmModal = _ConfirmModal2.default;
exports.ModalRoot = _ModalRoot2.default;
exports.modalReducer = _reducer2.default;
exports.default = _component2.default;
