// @evercourse/ever-modal/build/selectors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.getAllModals = exports.isConfirm = exports.dirtyFormExists = exports.isOpen = exports.getModalProps = exports.getModal = undefined;

var _reduxForm = require("redux-form");

var getModal = exports.getModal = function getModal(state, modalType) {
    return state.modals[modalType];
};

var getModalProps = exports.getModalProps = function getModalProps(state, modalType) {
    var modal = getModal(state, modalType);
    return modal && modal.modalProps;
};

var isOpen = exports.isOpen = function isOpen(state, modalType) {
    var modal = getModal(state, modalType);
    return modal && modal.visible;
};
var dirtyFormExists = exports.dirtyFormExists = function dirtyFormExists(state, modalType) {
    var props = getModalProps(state, modalType);
    return props && props.formName && (0, _reduxForm.isDirty)(props.formName)(state);
};

var isConfirm = exports.isConfirm = function isConfirm(state, modalType) {
    var modal = getModal(state, modalType);
    return modal && modal.modalType === "CONFIRM_MODAL";
};

var getAllModals = exports.getAllModals = function getAllModals(state) {
    return state.modals;
};
